File: backend/internal/models/game.go
type Position struct {
	X int `json:"x"`
	Y int `json:"y"`
}

type Player struct {
	ID             string   `json:"id"`
	Position       Position `json:"position"`
	Health         int      `json:"health"`
	ActionPoints   int      `json:"actionPoints"`
	MovementPoints int      `json:"movementPoints"`
	IsCurrentTurn  bool     `json:"isCurrentTurn"`
}

type GameState struct {
	Players     map[string]*Player `json:"players"`
	CurrentTurn string             `json:"currentTurn"`
	TurnOrder   []string           `json:"turnOrder"`
	TurnNumber  int                `json:"turnNumber"`
	GameStatus  string             `json:"gameStatus"`
}

File: frontend/src/components/Game/Grid/types/game.ts
export interface Position {
  x: number;
  y: number;
}

export interface Character {
  position: Position;
  color: string;
  symbol: string;
  movementPoints: number;
}

export interface Player {
  id: string;
  name: string;
  character: Character;
  isCurrentTurn: boolean;
}

export interface Players {
  [userId: string]: Player;
}
